spa/src/pages/ConfigSteps/index.tsx

```diff
@@ -55,31 +56,82 @@ const InlineDialog = styled(TooltipPrimitive)`
 	position: absolute;
 	top: -22px;
 `;
-
+const LoggedInContent = styled.div`
+	display: flex;
+	justify-content: center;
+	align-items: center;
+`;
 const ConfigSteps = () => {
-	const [completedStep1, setCompletedStep1] = useState(false);
+	const { username, email } = OAuthManagerInstance.getUserDetails();
+	const isAuthenticated = !!(username && email);
+
+	const originalUrl = window.location.origin;
+	const [selectedOption, setSelectedOption] = useState(0);
+
+	const [completedStep1, setCompletedStep1] = useState(isAuthenticated);
 	const [completedStep2] = useState(false);
+
 	const [showStep2, setShowStep2] = useState(true);
-	const [canViewContentForStep2, setCanViewContentForStep2] = useState(false);
-	const [selectedOption, setSelectedOption] = useState(0);
+	const [canViewContentForStep2, setCanViewContentForStep2] = useState(isAuthenticated);
+
+	const [expandStep1, setExpandStep1] = useState(!isAuthenticated);
+	const [expandStep2, setExpandStep2] = useState(isAuthenticated);
 
-	const authorize = () => {
+	const [isLoggedIn, setIsLoggedIn] = useState(isAuthenticated);
+	const [loggedInUser, setLoggedInUser] = useState<string>(username);
+	const [loaderForLogin, setLoaderForLogin] = useState(false);
+
+	useEffect(() => {
+		window.addEventListener("message", (event) => {
+			if (event.origin !== originalUrl) return;
+			OAuthManagerInstance.setTokens(event.data.accessToken, event.data.refreshToken);
+			setIsLoggedIn(true);
+			setCompletedStep1(true);
+			setExpandStep1(false);
+			setExpandStep2(true);
+			setCanViewContentForStep2(true);
+		});
+	}, []);
+
+	useEffect(() => {
+		OAuthManagerInstance.checkValidity().then((status: boolean | undefined) => {
```

Comment:
Write the function as async and do normal async code here? Instead of Promise chain.

Reply:
Using `async-await` inside useEffect causes [memory leakage](https://ultimatecourses.com/blog/using-async-await-inside-react-use-effect-hook). That's why I'm using a then chain here.